### DISCUSSION TITLE
How to define generalization and specialization relationships

### DISCUSSION THREAD

**Discussion by Disura-Randunu** [directus/directus#7938] [Archive]
How can we implement generalization or specialization relationships in directus. For example, someone or something who interacts with the system can be specified as an Interactor. An Interactor can be either a human or a logical entity. These human and logical entities store different information. How can we define a type field in the Interactor Entity to be selected as one of either Human or Logical entity?

**Comment by rijkvanzanten** [comment]
I'll have to be honest in that I'm a little lost in your specific example, but I believe what you're looking for is the Many-to-Any relationship type 👍🏻

**Reply by Disura-Randunu** [reply]
@rijkvanzanten Thanks for the reply. I have this weird requirement of having flexible relationships. That's why it's a bit confusing. Anyway, I just want a type field in one table which has some selectable values which point to some other tables. Sort of like a lookup table.

**Reply by rijkvanzanten** [reply]
Ah yes, that is the many to any relationship type. It allows you to setup a relationship where one table can point to multiple different other tables 👍🏻

**Reply by Disura-Randunu** [reply]
@rijkvanzanten I think I got it working. You were right, I was able to do this with Many-To-Any relationships. Thank you very much!!!
